WordPress/gutenberg
# WIP: [Block Library - Cover]: Add opacity to every possible background

## Description
Resolves: https://github.com/WordPress/gutenberg/issues/32339

This PR tries to add `opacity` handling in `Cover` block with any background (image/video or bacgroundColor/gradient). I've approached this by keeping the existing `span.wp-block-cover__gradient-background` and showing it always if `dimRatio` is applied - which should be always the case as we have a default value for this attribute.

I've removed some styles that were handling the opacity and added the opacity right into this `span` overlay layer. This way it makes nested `Cover` blocks appear as expected.


## Testing instructions
1. Make sure deprecated blocks appear as before
2. Insert `Cover` blocks and try different `opacity` settings and combinations.


## Notes
1. I have not updated yet the native files and might need some help there from the mobile team
2. I noticed in `deprecated` versions some extra attributes that didn't make sense like `contentAlign and title` where I removed them from some deprecations. This seemed to me that they were there only by some `copy/paste` of attributes in some new deprecations.
3. Some styles might need to be tweaked or removed. --cc @jasmussen 😄 
4. --Edit: I've just noticed that I don't handle well the case of an existing block with no overlay set in the front-end - it's not black.. So I'll check it out..


## Screenshots 
<img width="639" alt="Screenshot 2021-08-06 at 4 02 51 PM" src="https://user-images.githubusercontent.com/16275880/128514495-4ac17250-de60-4d60-98ee-3411c12c69a4.png">

--cc @justintadlock 
## Checklist:
